### ISSUE TITLE
Source generator

### ISSUE THREAD

**Issue by Nemanja986** [sveinungf/spreadcheetah#31]
Is there a way to use cell styles attributes with source generator?
For example, I want to highlight specific property in my class:

```
public class MyObject
{
        [HighlightExcelCell('some color')]
        public string Property{ get; set; }
}
```

**Comment by sveinungf** [comment]
This is not supported yet. You want to have all cells in a column with the same color? Or would different rows have different styling?

**Comment by Nemanja986** [comment]
I would like to color a specific cell for now, meaning all cells in a column the same color, but it would be good if both options were available, also, is it possible to limit the length of a value in a cell?
```
public class MyObject
{
        [[StringLength(30)]
        public string Property{ get; set; }
}
```
When can we expect version 1.13.0?

**Comment by sveinungf** [comment]
> I would like to color a specific cell for now, meaning all cells in a column the same color, but it would be good if both options were available.

Having the same color for all cells in a column could be done with an attribute like you've suggested. Having different colors depending on the row is not that straight-forward with the use of an attribute. If you have an idea how that could look like then please let me know! I want to make sure there is a somewhat similar approach that could be used for both use cases before implementing the same-color use case.

> Also, is it possible to limit the length of a value in a cell?

It's not currently possible, but I've created a separate issue to track this: https://github.com/sveinungf/spreadcheetah/issues/33

> When can we expect version 1.13.0?

Most likely when https://github.com/sveinungf/spreadcheetah/issues/21 has been implemented. It is currently work-in-progress. No ETA to share at this time.

**Comment by Nemanja986** [comment]
I agree that implementing different colors depending on the row isn't straight-forward. However, we could consider an attribute that colors the cell in a specific way based on the cell's value. For example, if the property is an integer and its value is greater than 20, I want to color that cell in red, something like that, thoughts?

In any case, these two attributes are quite okay for now (StringLength and HighlightExcelCell)

By the way, thank you for the quick response, and I must compliment you, excellent library.
